src/pool/fifo_wait.c

```diff
@@ -132,6 +134,65 @@ static void pool_push(ABT_pool pool, ABT_unit unit)
     pthread_mutex_unlock(&p_data->mutex);
 }
 
+static ABT_unit pool_pop_wait(ABT_pool pool, double time_secs)
+{
+    ABTI_pool *p_pool = ABTI_pool_get_ptr(pool);
+    data_t *p_data = pool_get_data_ptr(p_pool->data);
+    unit_t *p_unit = NULL;
+    ABT_unit h_unit = ABT_UNIT_NULL;
+
+    pthread_mutex_lock(&p_data->mutex);
+
+    if (!p_data->num_units) {
+#if defined(ABT_CONFIG_USE_CLOCK_GETTIME)
```

Comment:
Just for my own education: on what platforms might we expect ABT_CONFIG_USE_CLOCK_GETTIME to be undefined?  I like that there is a fallback here, but it would be good to know when it might happen because there will be a performance implication.

Reply:
I believe `ABT_CONFIG_USE_CLOCK_GETTIME` is enabled on 100% (or almost 100%) platforms. This branch is to make sure Argobots can be compiled in a special case.

This check comes from https://github.com/pmodels/argobots/blob/main/src/arch/abtd_time.c , which I believe should stem from MPICH's code (`MPI_Wtime()`), but I could not find platforms that do not support `clock_gettime()` (except for Windows, which Argobots does not support).

If such a platform would exist, there should be a way to get an absolute time for `pthread_cond_timedwait()`, so once we find such, we will fix this implementation.

Reply:
Ah Ok.  Glad to hear that it's not expected to be a common scenario.   Thanks!